pkg/controller/networkpolicy/validate.go

```diff
@@ -145,21 +145,22 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(op admv1.Operation, tier s
 }
 
 // validateRuleName validates if the name of each rule is unique within a policy
-func (v *NetworkPolicyValidator) validateRuleName(ingress, egress []secv1alpha1.Rule) bool {
+func (v *NetworkPolicyValidator) validateRuleName(rules []secv1alpha1.Rule) (bool, string) {
 	uniqueRuleName := sets.NewString()
-	isUnique := func(rules []secv1alpha1.Rule) bool {
-		for _, rule := range rules {
-			if uniqueRuleName.Has(rule.Name) {
-				return false
-			}
-			if rule.Name != "" {
-				uniqueRuleName.Insert(rule.Name)
-			}
+	for _, rule := range rules {
+		// If a rule doesn't have a name, it means that
+		// mutatingWebhook skip its generation due to illegal input
+		if rule.Name == "" {
+			return false, "please check if you have duplicate rules or one of your rule-names contains hash value of other rules, which is illegal"
```

Comment:
```suggestion
			return false, "policy has duplicate rules, or collision occurs in generated rule names"
```
Also we need to document to rule name auto-gen process and reason for potential collision in `antrea-network-policy.md`

Reply:
Thanks!